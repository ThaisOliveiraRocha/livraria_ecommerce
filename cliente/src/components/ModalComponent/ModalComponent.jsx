import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  ContainerModal,
  HeaderModal,
  BodyModal,
  Overlay,
  FooterModal
} from "./ModalComponent.styles";
import Titulo from "../Title";
import Texto from "../Text";
import Button from "../Button";
import { withRouter } from "react-router-dom";

export const ModalComponent = ({
  temaModal,
  mensagem,
  funcaoConfirma,
  cancel
}) => {
  return (
    <>
      <Overlay onClick={cancel} />
      <ContainerModal>
        <Button onClick={cancel} className="close" background="#818A96">
          <Texto color="white">X</Texto>
        </Button>
        <HeaderModal>
          <Titulo>{temaModal}</Titulo>
        </HeaderModal>
        <BodyModal>
          <Texto>{mensagem}</Texto>
        </BodyModal>
        <FooterModal>
          <Button
            onClick={funcaoConfirma}
            className="margin-right-10 btnConfirmar"
          >
            <Texto color="white">Confirmar</Texto>
          </Button>
          <Button background="#818A96" onClick={cancel} className="btnCancelar">
            <Texto color="white">Cancelar</Texto>
          </Button>
        </FooterModal>
      </ContainerModal>
    </>
  );
};

ModalComponent.propTypes = {
  // bla: PropTypes.string,
};

ModalComponent.defaultProps = {
  // bla: 'test',
};

const mapStateToProps = state => ({
  // blabla: state.blabla,
});

export default withRouter(connect(mapStateToProps, null)(ModalComponent));
