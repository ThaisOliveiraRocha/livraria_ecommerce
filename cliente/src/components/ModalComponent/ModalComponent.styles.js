import styled from "styled-components";

export const ContainerModal = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  border: 1px solid rgba(11, 107, 230, 0.5);
  border-radius: 5px;
  box-shadow: 5px 5px 30px rgba(11, 107, 230, 0.3);
  background-color: white;
  z-index: 15;

  position: absolute;
  left: 40.7%;
  top: 20%;
  padding: 20px;
`;

export const HeaderModal = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const BodyModal = styled.div`
  display: flex;
  align-items: center;

  padding: 20px;
`;

export const FooterModal = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Overlay = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 10;
`;
