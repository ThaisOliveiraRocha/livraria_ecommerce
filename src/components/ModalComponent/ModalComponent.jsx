import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  ContainerModal,
  HeaderModal,
  BodyModal,
  Overlay,
  FooterModal
} from "./ModalComponent.styles";
import Titulo from "../Title";
import Texto from "../Text";
import Button from "../Button";

const ModalComponent = ({ temaModal, mensagem, funcaoConfirma, cancel }) => {
  return (
    <>
      <Overlay onClick={cancel} />
      <ContainerModal>
        <Button onClick={cancel} className="close">
          X
        </Button>
        <HeaderModal>
          <Titulo>{temaModal}</Titulo>
        </HeaderModal>
        <BodyModal>
          <Texto>{mensagem}</Texto>
        </BodyModal>
        <FooterModal>
          <Button
            background="green"
            onClick={funcaoConfirma}
            className="margin-right-10"
          >
            Confirmar
          </Button>
          <Button background="red" onClick={cancel}>
            Cancelar
          </Button>
        </FooterModal>
      </ContainerModal>
    </>
  );
};

ModalComponent.propTypes = {
  // bla: PropTypes.string,
};

ModalComponent.defaultProps = {
  // bla: 'test',
};

const mapStateToProps = state => ({
  // blabla: state.blabla,
});

export default connect(mapStateToProps, null)(ModalComponent);
