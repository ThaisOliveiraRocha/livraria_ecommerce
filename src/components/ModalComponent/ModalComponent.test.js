import React from "react";
import { shallow } from "enzyme";
import ModalComponent from "./ModalComponent";

describe("<ModalComponent />", () => {
  it("should render without crashing", () => {
    shallow(<ModalComponent />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<ModalComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
