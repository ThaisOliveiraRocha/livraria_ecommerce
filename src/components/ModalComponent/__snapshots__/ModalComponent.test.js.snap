// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalComponent /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
